File: Heat/Controllers/SearchController.vb
Imports System.Web.Mvc

Namespace Controllers
    Public Class SearchController
        Inherits Controller

        <HttpPost> _
        Function Index(searchquery As String, fromcontroller As String) As ActionResult
            Select Case fromcontroller
                Case "Customers"
                    ViewBag.message = "Search results for " & searchquery & " in Customers"
                Case "PlantClasses"
                    ViewBag.message = "Search results for " & searchquery & " in  PlantClasses"
                Case "Plants"
                    ViewBag.message = "Search results for " & searchquery & " in  Plant"
                Case "PlantTypes"
                    ViewBag.message = "Search results for " & searchquery & " in  PlantTypes"

            End Select
            Return View()
        End Function
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/SearchController.cs
using System.Web.Mvc;

namespace Heat.Controllers
{
    [Authorize()]
	public class SearchController : Controller
	{

		[HttpPost()]
		public ActionResult Index(string searchquery, string fromcontroller)
		{
			switch (fromcontroller) {
				case "Customers":
					ViewBag.message = "Search results for " + searchquery + " in Customers";
					break;
				case "PlantClasses":
					ViewBag.message = "Search results for " + searchquery + " in  PlantClasses";
					break;
				case "Plants":
					ViewBag.message = "Search results for " + searchquery + " in  Plant";
					break;
				case "PlantTypes":
					ViewBag.message = "Search results for " + searchquery + " in  PlantTypes";

					break;
			}
			return View();
		}
	}
}
